base/array.jl

```diff
@@ -1684,6 +1684,28 @@ end
 findfirst(testf::Function, A::Union{AbstractArray, AbstractString}) =
     findnext(testf, A, first(keys(A)))
 
+"""
+    isfound(needle, haystack)
+
+Determine whether `needle` is found within `haystack`. This function is effectively
+equivalent to `findfirst(needle, haystack) !== nothing`.
+
+See also [`findfirst`](@ref) and [`any`](@ref).
+
+# Examples
+```jldoctest
+julia> isfound('x', "hexonxonx")
```

Comment:
This currently leads to a deprecation warning and an error. I think it would make sense to add a method `isfound(::Char, ::AbstractString)` to handle this.

Reply:
Good catch. Though for `Char` containment in `String` should we just encourage using `in`? Having `isfound(scalar, collection)` work when its documented fallback, `findfirst(scalar, collection) !== nothing`, does not work is a little weird.

Reply:
I believe the only reason `findfirst` doesn't support that is due to a deprecation, but that case would not have been useful (looking for a string in a char; yes you read that right). So we could add the method to `findfirst` as well, either now or after 0.7 if we want to be super conservative.

Reply:
IIRC I didn't add `findfirst(::Char, ::String)` just because it wasn't strictly necessary (i.e. the standard pattern for general collections based on `equalto` works well), and because can always be added later. At least that's what I noted in the description of https://github.com/JuliaLang/julia/pull/24673. OTOH the `findfirst(::String ::String)` method is needed because there is no generic API to looking for a subsequence in general collections at the moment.

It could make sense to add a special case for `findfirst(::Char, ::String)` for convenience, though, now that the basic API has stabilized.

Reply:
Okay, so action items here:

* Add `findfirst(::Char, ::AbstractString)`, which facilitates a corresponding `isfound` method
* Add back `contains(::AbstractString, ::Union{Char,AbstractString})`

Currently all methods for `contains` deal with strings and regular expressions. So effectively rather than _replacing_ `contains` with `isfound`, which this PR currently does, we'll _add_ `isfound` and replace the `Regex` methods for `contains` with methods for `isfound`. That is, deprecate `contains(::AbstractString, ::Regex)` in favor of `isfound` and otherwise leave `contains` alone.

Is that right?

Reply:
+1 for the first bullet, but do we really need to keep `contains`? Is that really what you suggested @JeffBezanson?

Reply:
Okay, so updated action items:

* Add `findfirst(::Char, ::AbstractString)`
* Revert deprecation of `contains`

So the PR would effectively be what it originally was, just adding `isfound` and deprecating callable regexes. Is that right?

Reply:
I meant to go ahead with removing `contains` as agreed, and separately consider whether we want `contains(string, string)` as a convenience function.

Reply:
Okay, so just add `findfirst(::Char, ::AbstractString)` and leave the rest of the PR as-is.